lib/repositories/user_repository.dart

```diff
@@ -2,14 +2,19 @@ import 'package:injectable/injectable.dart';
 
 import 'package:shop_app/models/index.dart';
 import 'package:shop_app/services/http/http_client.dart';
+import 'package:shop_app/services/index.dart';
 
 @Injectable(scope: 'auth')
 class UserRepository {
   final HttpClient httpClient;
 
   UserRepository(this.httpClient);
 
-  Future<UserProfile> getUserProfile() async {
-    return UserProfile.fromJson({'userName': 'test'});
+  String get endpointUrl => '/users';
+
+  Future<UserProfile> getUserProfileById(int id) async {
+    final response = await httpClient.get<DynamicMap>('$endpointUrl/$id');
```

Comment:
It's no need to add generic types

Reply:
Should I cast it later? Because linter is arguing)